marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/entities/entity-search-lib.sjs

```diff
@@ -43,10 +43,13 @@ function addPropertiesToSearchResponse(entityName, searchResponse, propertiesToD
     selectedPropertyMetadata = buildSelectedPropertiesMetadata(allMetadata, selectedPropertyNames);
   }
   selectedPropertyMetadata = selectedPropertyMetadata.length > 0 ? selectedPropertyMetadata : propertyMetadata.slice(0, maxDefaultProperties);
+  let selectedPropertyMetadataFinal = JSON.parse(JSON.stringify(selectedPropertyMetadata));
```

Comment:
I'm unclear why it is stringifying and then parsing immediately after. Is it is to convert from a JSON node to a plain JSON object? If so, I think that is worth adding a comment for. Also,  `selectedPropertyMetadata.toObject()` may be a better way.

Reply:
This is to create a deep copy of selectedPropertyMetadata with its own object references, as I later modify the objects inside of selectedPropertyMetadataFinal. I will add a clarifying comment.

Reply:
Is this no longer needed now that selectedPropertyMetadata isn't being modified?